src/build/arg/mod.rs

```diff
@@ -5174,7 +5174,10 @@ mod test {
 
     #[test]
     fn positional_display_required() {
-        let p2 = Arg::new("pos").index(1).setting(ArgSettings::Required);
+        let p2 = Arg::new("pos")
+            .index(1)
+            .setting(ArgSettings::Required)
+            .setting(ArgSettings::TakesValue);
```

Comment:
Oh, `Display` is for `Arg`.  If `Display` is used before `_build` is called, then `TakesValue` is not set.

Maybe we need to revert this so `Display` is always valid?  Or add a helper, like `is_positional` for `TakesValue`?